templates/commands/goldentest/record.go

```diff
@@ -60,9 +63,37 @@ func (c *RecordCommand) Run(ctx context.Context, args []string) error {
 		return fmt.Errorf("failed to parse flags: %w", err)
 	}
 
-	if _, err := parseTestCases(c.flags.Location, c.flags.TestName); err != nil {
+	testCases, err := parseTestCases(c.flags.Location, c.flags.TestName)
+	if err != nil {
 		return fmt.Errorf("failed to parse golden test: %w", err)
 	}
 
-	return fmt.Errorf("unimplemented")
+	// Create a temporary directory to validate golden tests rendered with no
+	// error. If any test fails, no data should be written to file system
+	// for atomicity purpose.
+	tempDir, err := getTempDir()
+	if err != nil {
+		return fmt.Errorf("failed to create temporary directory: %w", err)
+	}
+	var merr error
+	for _, tc := range testCases {
+		merr = errors.Join(merr, renderTestCase(c.flags.Location, tempDir, tc))
+	}
+	if merr != nil {
+		return fmt.Errorf("failed to render golden tests: %w", merr)
+	}
+	if err := os.RemoveAll(tempDir); err != nil {
+		return fmt.Errorf("failed to remove temporary directory: %w", err)
+	}
+
+	// Render outputs in template golden test directories.
+	logger := logging.DefaultLogger()
```

Comment:
Perhaps the logging style should be consistent with other places in the abc cli, see https://github.com/search?q=repo%3Aabcxyz%2Fabc%20logging&type=code

Reply:
Done.